File: Circuits/DEC18/Picking coins.cpp
#include<bits/stdc++.h>
#define ull unsigned long long int 
using namespace std;
int main()
{
    int T,i;	cin>>T;
	while(T--)
	{
		ull n,k;
		cin>>n>>k;
		ull m  = 0,t = k;
		if(k == 1)
		    n = n%2;
		while(n>= 1LL*2*k)
		{
			n-=(1LL*2*k);
			k = 1LL * k * t;
		}
		if(n>=k)	
			m=1;
		if(m%2)		cout<<"Alice"<<endl;
		else		cout<<"Bob"<<endl;
	}
}

File: Circuits/DEC18/pickingCoins.py
for _ in range(int(input())):
	s = list(map(int,input().split()))
	n = s[0]
	k = s[1]
	m = 0
	t = k
	while n >= 2*k :
		n = n - 2 * k
		k = k * t
	if n > k : 
	       m = 1
	if m%2 == 1:
		print("Alice")
	else:
		print("Bob")